traffic_ops/traffic_ops_golang/deliveryservice/letsencrypt_dns_challenge.go

```diff
@@ -0,0 +1,73 @@
+package deliveryservice
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"errors"
+	"net/http"
+
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+)
+
+type DnsRecord struct {
+	Fqdn   *string `json:"fqdn" db:"fqdn"`
+	Record *string `json:"record" db:"record"`
+}
+
+func GetDnsChallengeRecords(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	getQuery := `SELECT fqdn, record FROM dnschallenges`
+
+	if inf.Params["fqdn"] != "" {
+		getQuery += ` where fqdn = '` + inf.Params["fqdn"] + `'`
```

Comment:
This is dangerous. If I request `GET /api/1.4/letsencrypt/dnsrecords?fqdn=test'%3BDROP%20TABLE%20tm_user%20CASCADE%3B--` (URI Decoded: `GET /api/1.4/dnsrecords?fqdn=test';DROP TABLE tm_user CASCADE;--` I can destroy the entire user table, for example. You should use [`dbhelpers.BuildWhereAndOrderByAndPagination`](https://godoc.org/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/dbhelpers#BuildWhereAndOrderByAndPagination) to help build that for you safely. Plus, it gives you pagination and sorting controls basically for free.

Reply:
done